app/gui/view/graph-editor/src/builtin/visualization/java_script/warnings.js

```diff
@@ -0,0 +1,53 @@
+/** Simple Warning Visualization. */
+class WarningsVisualization extends Visualization {
+    static inputType = 'Any'
+    static label = 'Warnings'
+
+    constructor(data) {
+        super(data)
+        this.setPreprocessor('Standard.Visualization.Warnings', 'process_to_json_text')
+    }
+
+    onDataReceived(data) {
+        while (this.dom.firstChild) {
+            this.dom.removeChild(this.dom.lastChild)
+        }
+
+        const divElement = document.createElement('div')
+        divElement.setAttributeNS(null, 'id', 'vis-tbl-view')
+        divElement.setAttributeNS(null, 'class', 'scrollable')
+        divElement.setAttributeNS(null, 'width', '100%')
+        divElement.setAttributeNS(null, 'height', '100%')
+        this.divElem = divElement
+        this.dom.appendChild(divElement)
+        this.updateDivSize()
+
+        const parsedData = typeof data === 'string' ? JSON.parse(data) : data
+        const prefix = '<ul style="font-family: DejaVuSansMonoBook, sans-serif; font-size: 12px; white-space: pre;">'
+        const body = parsedData.map((d) => '<li>'+this.escapeHtml(d)+'</li>').join('')
+        const suffix = '</ul>'
```

Comment:
If there are no warnings, this will display an empty box which is hard to distinguish from, for example, visualization error.

Maybe we could add a header before the `<ul>`, like:
`There are {data.length} warnings:` or `There are no warnings.` if `data.length == 0`?

Reply:
Will add a default message if no warnings attached.

Reply:
![image](https://user-images.githubusercontent.com/4699705/212074254-fa0ed10d-b101-4419-b28e-c59412b39494.png)

Reply:
Perfect.